core/src/main/java/org/apache/druid/java/util/common/StringUtils.java

```diff
@@ -309,11 +311,54 @@ public static String emptyToNullNonDruidDataString(@Nullable String string)
    * Convert an input to base 64 and return the utf8 string of that byte array
    *
    * @param input The string to convert to base64
-   *
    * @return the base64 of the input in string form
    */
   public static String utf8Base64(String input)
   {
-    return fromUtf8(Base64.getEncoder().encode(toUtf8(input)));
+    return fromUtf8(encodeBase64(toUtf8(input)));
+  }
+
+  /**
+   * Convert an input byte array into a newly-allocated byte array using the {@link Base64} encoding scheme
+   *
+   * @param input The byte array to convert to base64
+   * @return the base64 of the input in byte array form
+   */
+  public static byte[] encodeBase64(byte[] input)
+  {
+    return BASE64_ENCODER.encode(input);
+  }
+
+  /**
+   * Convert an input byte array into a string using the {@link Base64} encoding scheme
+   *
+   * @param input The byte array to convert to base64
+   * @return the base64 of the input in string form
+   */
+  public static String encodeBase64ToString(byte[] input)
```

Comment:
Nit: I think `encodeToBase64String` would read a bit more smoothly and more like the java.util version `encodeToString` than the current `encodeBase64ToString`. It would also be nice if it matched the other encode method, so maybe change it to `encodeToBase64`, or change this one to `encodeBase64String` and leave the other as is.

Likewise, calling one of the decode methods out explicitly as the string one would be nice, maybe `decodeBase64` and `decodeBase64String`?

Reply:
Hi, @clintropolis . Thanks for your comment. Indeed, `encodeBase64String` would be better.

> Likewise, calling one of the decode methods out explicitly as the string one would be nice, maybe decodeBase64 and decodeBase64String?

Hmmm... About this, do you mean code implementation like the following?

```java
public static String decodeBase64String(byte[] input)
{
  return fromUtf8(BASE64_DECODER.decode(input));
}
```

Reply:
Oh for the decode methods, i mean to distinguish the one that takes a `String` argument from the one that takes a `byte[]` argument

Reply:
I see. I feel the same. Thx.